mat64/format.go

```diff
@@ -9,15 +9,66 @@ import (
 	"strconv"
 )
 
-// Format prints a pretty representation of m to the fs io.Writer. The format character c
+// Formatted returns a fmt.Formatter for the matrix m using the given options.
+func Formatted(m Matrix, options ...FormatOption) fmt.Formatter {
+	f := formatter{
+		matrix: m,
+		dot:    '.',
+	}
+	for _, o := range options {
+		o(&f)
+	}
+	return f
+}
+
+type formatter struct {
+	matrix Matrix
+	prefix string
+	margin int
+	dot    byte
+}
+
+// FormatOption is a functional option for matrix formatting.
+type FormatOption func(*formatter)
+
+// Prefix sets the formatted prefix to the string p. Prefix is a string that is prepended to
+// each line of output.
+func Prefix(p string) FormatOption {
+	return func(f *formatter) { f.prefix = p }
+}
+
+// Margin sets the format margin to m. If m is greater than zero, the margin option indicates
```

Comment:
Edge is nice, but unfortunately overloaded and doesn't cover all the behaviours (damn English); it may print the ends rather than the edges, not all the edge is printed, and if the matrix is small enough all is printed.